sdk/storage/azure-storage-internal-avro/src/main/java/com/azure/storage/internal/avro/implementation/AvroParserState.java

```diff
@@ -0,0 +1,130 @@
+package com.azure.storage.internal.avro.implementation;
+
+import com.azure.storage.internal.avro.implementation.schema.AvroSchema;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Stack;
+
+/**
+ * Represents the current state of the AvroParser
+ *
+ * This class internally holds state to keep track of
+ *
+ * <li> the data available to the parser. </li>
+ * <li> a  stack of objects that need to be parsed. </li>
+ * <li> the current size of the cached buffers. </li>
+ */
+public class AvroParserState {
+
+    private Stack<AvroSchema> stack;
+    private List<ByteBuffer> cache;
+    private long size;
+
+    /**
+     * Creates a new instance of an AvroParserState.
+     */
+    AvroParserState() {
+        this.stack = new Stack<>();
+        this.cache = new LinkedList<>();
+        this.size = 0;
+    }
+
+    /**
+     * Writes to the state's internal cache of buffers.
+     *
+     * @param buffer The buffer to add to the cache.
+     */
+    void write(ByteBuffer buffer) {
```

Comment:
Wondering if naming these like writeBuffer (cacheBuffer?), pushSchema(), peekSchema() would help make clear that these objects are being tracked differently? Although arguably I guess the types of the parameters/return values do that for you.

Reply:
Yeah I can change them to be more descriptive. It may help improve readability.